documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -1,20 +1,36 @@
 Pass Data to Algorithms
 #######################
 
-When using the C++ standard execution policies, oneDPL supports data being passed to its algorithms as specified
-in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard, the calling code
-must prevent data races when using algorithms with parallel execution policies.
+When using the C++ standard aligned (or *host*) execution policies, oneDPL supports data being passed to its
+algorithms as specified in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard,
+the calling code must prevent data races when using algorithms with parallel execution policies.
 
-Note: Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
-of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
-for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
-input with the C++ standard execution policies.
+.. note::
+   Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
+   of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
+   for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
+   input with the C++ standard aligned execution policies.
 
 When using a device execution policy, you can use one of the following ways to pass data to an algorithm:
 
-* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions
-* Unified shared memory (USM) pointers
-* ``std::vector`` with or without a USM allocator
+* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions for a `SYCL buffer
+  <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:buffers>`_;
+* `Unified shared memory (USM) <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:usm>`_ pointers;
+* ``std::vector`` with or without a USM allocator.
+
+For an algorithm to access data, it is important that the used execution policy matches the data storage type.
+The following table shows which execution policies can be used with various data storage types.
+
+====================================== =============== =============
+Data Storage                           Device Policies Host Policies
+====================================== =============== =============
+SYCL buffer                            Yes             No
+Device-allocated USM                   Yes             No
+Shared and host-allocated USM          Yes             Yes
+``std::vector`` with a USM allocator   Yes             Yes
+``std::vector`` with a host allocator  See below       Yes
```

Comment:
```std::vector`` with a host allocator See below`
1) "See below" - there is no link where I should "see below".
2) ````std::vector`` with a host allocator` confuses a little bit. Does it mean any host allocator? i that case it is "general C++ host  code" case... Probably, we can dont remind about that (?)

Reply:
Are there problems with supporting "any host allocator"? For as long as the iterator is not a pointer, data copying will be used, right? The "Use std::vector" section in this very document (see below :)) does not specify details of host allocators either.

Reply:
Added the link.